applications/GeoMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp

```diff
@@ -99,4 +99,38 @@ Matrix ConstitutiveLawUtilities::MakeInterfaceConstitutiveMatrix(double      Nor
     return result;
 }
 
+void ConstitutiveLawUtilities::CheckStrainSize(const Properties&               rProperties,
+                                               const std::vector<std::size_t>& rExpectedStrainSizes,
+                                               std::size_t                     Dim,
+                                               std::size_t                     ElementId)
+{
+    const std::size_t strain_size = rProperties[CONSTITUTIVE_LAW]->GetStrainSize();
+    const auto        size_exists =
+        std::any_of(rExpectedStrainSizes.begin(), rExpectedStrainSizes.end(),
+                    [strain_size](std::size_t expected_size) { return strain_size == expected_size; });
+    if (!size_exists) {
+        std::stringstream ss;
+        std::copy(rExpectedStrainSizes.begin(), rExpectedStrainSizes.end(),
+                  std::ostream_iterator<std::size_t>(ss, " "));
+        KRATOS_ERROR << "Wrong constitutive law used. This is a " << Dim << "D element! Expected strain size is "
+                     << ss.str() << " (element Id = " << ElementId << ")." << std::endl;
+    }
+}
+
+void ConstitutiveLawUtilities::CheckAvailabilityOfStrainMeasure_Infinitesimal(const Properties& rProperties,
```

Comment:
If I read what the function does, the following name (or something comparable) would more clearly express its function (but that's more a personal preference, so feel free to keep it as is, if you think that's more clear):
```suggestion
void ConstitutiveLawUtilities::CheckConstitutiveLawHasStrainMeasure_Infinitesimal(const Properties& rProperties,
```

Reply:
Yes, your version is clearer. However, I skip ConstitutiveLaw in the name because by now it is called with `ConstitutiveLawUtilities::`. Is it acceptable? If someone will add `using `then there is no need for it.

Reply:
Yes, thanks!